test/gtest/uct/test_md.cc

```diff
@@ -307,23 +307,33 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
                               UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH;
 
         for (unsigned i = 0; i < 300; i++) {
-            offset        = ucs::rand() % buffer_size;
-            length_offset = ucs::rand() % buffer_size;
+            slice_offset = ucs::rand() % buffer_size;
+            slice_length = ucs::rand() % buffer_size;
+
+            if (!slice_length) {
+                continue;
+            }
 
             status = uct_md_mem_query(md(),
-                                      UCS_PTR_BYTE_OFFSET(address, offset),
-                                      length_offset, &mem_attr);
+                                      UCS_PTR_BYTE_OFFSET(address,
+                                                          slice_offset),
+                                      slice_length, &mem_attr);
             ASSERT_UCS_OK(status);
             EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
             EXPECT_TRUE((mem_attr.alloc_length == buffer_size) ||
-                        (mem_attr.alloc_length == length_offset));
+                        (mem_attr.alloc_length == slice_length));
             EXPECT_TRUE((mem_attr.base_address ==
-                         UCS_PTR_BYTE_OFFSET(address, offset)) ||
+                         UCS_PTR_BYTE_OFFSET(address, slice_offset)) ||
```

Comment:
if mem-type is cuda:
EXPECT_EQ(address, mem_attr.base_address)
else:
EXPECT_EQ(UCS_PTR_BYTE_OFFSET(address, slice_offset), mem_attr.base_address)